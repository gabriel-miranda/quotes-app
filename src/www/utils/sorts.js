const byAuthor = 'authorName';
const byText = 'text';

export default {
  [byAuthor]: 'Author A-Z',
  [`-${byAuthor}`]: 'Author Z-A',
  [byText]: 'Text A-Z',
  [`-${byText}`]: 'Text Z-A',
};
