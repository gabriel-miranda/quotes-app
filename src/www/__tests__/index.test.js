/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Index from '../pages/index';

import page1 from './__mocks__/page1.mock';
import page2 from './__mocks__/page2.mock';

import sorts from '../utils/sorts';

const testError = new Error('Test error');

describe('Index test with enzyme', () => {
  it('should load the quotes properly in the state', async () => {
    const index = shallow(<Index data={page1} error={null} />);
    expect(index.state().data.results.length).toBeGreaterThan(0);
  });

  it('should load the new quotes in the state', async () => {
    const index = shallow(<Index data={page1} error={null} />);
    index.instance().addNewPage(page2);
    expect(index.update().state().data.results.length).toBeGreaterThan(10);
  });

  it('should load the next page in the state', async () => {
    const index = shallow(<Index data={page1} error={null} />);
    index.instance().addNewPage(page2);
    expect(index.update().state().data.pagination.page).toEqual(2);
  });

  it('should load error in state if an error happend', async () => {
    const index = shallow(<Index data={page1} error={testError} />);
    expect(index.update().state().error).toEqual(testError);
  });

  it('should change sort when calling handleChangeSort with a sortBy', async () => {
    const index = shallow(<Index data={page1} error={testError} />);
    const sortBy = Object.keys(sorts)[1];
    index.instance().handleChangeSort(sortBy);
    expect(index.update().state().sortBy).toEqual(sortBy);
  });

  it('should change searchQuery when calling handleSearch with parameters', async () => {
    const index = shallow(<Index data={page1} error={null} />);
    index.instance().handleSearch('text', 'miracle');
    expect(index.update().state().searchQuery).toMatchObject({field: 'text', text: 'miracle'});
  });

  it('should not change state when calling handleSearch without text', async () => {
    const index = shallow(<Index data={page1} error={null} />);
    const prevState = index.update().state();
    index.instance().handleSearch();
    expect(index.update().state()).toMatchObject(prevState);
  });

  it('should return empty query when calling buildQuery without setting query', async () => {
    const index = shallow(<Index data={page1} error={null} />);
    expect(index.instance().buildQuery()).toMatchObject({});
  });

  it('should reset sorting when making a search', async () => {
    const index = shallow(<Index data={page1} error={null} />);
    const sortBy = Object.keys(sorts)[1];
    index.instance().handleChangeSort(sortBy);
    index.instance().handleSearch('text', 'miracle');
    expect(index.update().state().sortBy).toEqual(Object.keys(sorts)[0]);
  });

  it('should empty the data when calling reset', async () => {
    const index = shallow(<Index data={page1} error={testError} />);
    const sortBy = Object.keys(sorts)[1];
    index.instance().reset(sortBy);
    expect(index.update().state().data).toEqual(null);
  });
});

describe('Index test with Snapshot Testing', () => {
  it('should show Index', () => {
    const component = renderer.create(<Index data={page1} error={null} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
