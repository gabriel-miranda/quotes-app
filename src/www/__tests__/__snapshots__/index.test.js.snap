// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index test with Snapshot Testing should show Index 1`] = `
Array [
  <header
    className="site-header"
    style={
      Object {
        "background": "#fff",
      }
    }
  >
    <nav
      className="navbar navbar-default"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-md-6 col-md-offset-3"
          >
            <div
              className="navbar-header"
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                  }
                }
              >
                <img
                  alt="quotes app logo"
                  src="/static/logo.png"
                  style={
                    Object {
                      "height": "42px",
                      "margin": "0 1em 0 0",
                    }
                  }
                />
                <h1
                  style={
                    Object {
                      "fontSize": "25px",
                      "margin": "0",
                    }
                  }
                >
                  Quotes app
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </header>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 col-md-offset-3"
      >
        <select
          className="form-control pull-right"
          onChange={[Function]}
          style={
            Object {
              "margin": "20px 0 0",
              "width": "auto",
            }
          }
        >
          <option
            value="authorName"
          >
            Author A-Z
          </option>
          <option
            value="-authorName"
          >
            Author Z-A
          </option>
          <option
            value="text"
          >
            Text A-Z
          </option>
          <option
            value="-text"
          >
            Text Z-A
          </option>
        </select>
      </div>
    </div>
  </div>,
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 col-md-offset-3"
      >
        <div
          className="card-docs"
          style={
            Object {
              "borderRadius": "3px",
              "margin": "20px 0",
              "padding": "20px",
            }
          }
        >
          <p>
            Quote 
            <strong>
              #
              91
            </strong>
          </p>
          <p>
            "
            It takes both sunshine and rain to make a rainbow.
            "
          </p>
          <p>
            <strong>
              -
              <em>
                Anonymous
              </em>
            </strong>
          </p>
        </div>
        <div
          className="card-docs"
          style={
            Object {
              "borderRadius": "3px",
              "margin": "20px 0",
              "padding": "20px",
            }
          }
        >
          <p>
            Quote 
            <strong>
              #
              92
            </strong>
          </p>
          <p>
            "
            A beautiful thing is never perfect.
            "
          </p>
          <p>
            <strong>
              -
              <em>
                Anonymous
              </em>
            </strong>
          </p>
        </div>
        <div
          className="card-docs"
          style={
            Object {
              "borderRadius": "3px",
              "margin": "20px 0",
              "padding": "20px",
            }
          }
        >
          <p>
            Quote 
            <strong>
              #
              37
            </strong>
          </p>
          <p>
            "
            From small beginnings come great things.
            "
          </p>
          <p>
            <strong>
              -
              <em>
                Anonymous
              </em>
            </strong>
          </p>
        </div>
        <div
          className="card-docs"
          style={
            Object {
              "borderRadius": "3px",
              "margin": "20px 0",
              "padding": "20px",
            }
          }
        >
          <p>
            Quote 
            <strong>
              #
              90
            </strong>
          </p>
          <p>
            "
            The day is already blessed, find peace within it.
            "
          </p>
          <p>
            <strong>
              -
              <em>
                Anonymous
              </em>
            </strong>
          </p>
        </div>
        <div
          className="card-docs"
          style={
            Object {
              "borderRadius": "3px",
              "margin": "20px 0",
              "padding": "20px",
            }
          }
        >
          <p>
            Quote 
            <strong>
              #
              16
            </strong>
          </p>
          <p>
            "
            Today is the tomorrow we worried about yesterday.
            "
          </p>
          <p>
            <strong>
              -
              <em>
                Anonymous
              </em>
            </strong>
          </p>
        </div>
        <div
          className="card-docs"
          style={
            Object {
              "borderRadius": "3px",
              "margin": "20px 0",
              "padding": "20px",
            }
          }
        >
          <p>
            Quote 
            <strong>
              #
              18
            </strong>
          </p>
          <p>
            "
            Every man dies. Not every man really lives.
            "
          </p>
          <p>
            <strong>
              -
              <em>
                Anonymous
              </em>
            </strong>
          </p>
        </div>
        <div
          className="card-docs"
          style={
            Object {
              "borderRadius": "3px",
              "margin": "20px 0",
              "padding": "20px",
            }
          }
        >
          <p>
            Quote 
            <strong>
              #
              28
            </strong>
          </p>
          <p>
            "
            The day is already blessed, find peace within it.
            "
          </p>
          <p>
            <strong>
              -
              <em>
                Anonymous
              </em>
            </strong>
          </p>
        </div>
        <div
          className="card-docs"
          style={
            Object {
              "borderRadius": "3px",
              "margin": "20px 0",
              "padding": "20px",
            }
          }
        >
          <p>
            Quote 
            <strong>
              #
              17
            </strong>
          </p>
          <p>
            "
            It's easier to see the mistakes on someone else's paper.
            "
          </p>
          <p>
            <strong>
              -
              <em>
                Anonymous
              </em>
            </strong>
          </p>
        </div>
        <div
          className="card-docs"
          style={
            Object {
              "borderRadius": "3px",
              "margin": "20px 0",
              "padding": "20px",
            }
          }
        >
          <p>
            Quote 
            <strong>
              #
              113
            </strong>
          </p>
          <p>
            "
            Some pursue happiness, others create it.
            "
          </p>
          <p>
            <strong>
              -
              <em>
                Anonymous
              </em>
            </strong>
          </p>
        </div>
        <div
          className="card-docs"
          style={
            Object {
              "borderRadius": "3px",
              "margin": "20px 0",
              "padding": "20px",
            }
          }
        >
          <p>
            Quote 
            <strong>
              #
              107
            </strong>
          </p>
          <p>
            "
            Every new day is another chance to change your life.
            "
          </p>
          <p>
            <strong>
              -
              <em>
                Anonymous
              </em>
            </strong>
          </p>
        </div>
        <div
          className="spinner spinner-lg is-auth0"
          style={
            Object {
              "margin": "0 auto 20px",
            }
          }
        >
          <div
            className="circle"
          />
        </div>
        <span
          style={
            Object {
              "fontSize": 0,
            }
          }
        />
      </div>
    </div>
  </div>,
]
`;
